src/csharp/Tokenizer.cs

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.ML.OnnxRuntimeGenAI
+{
+    public class Tokenizer : IDisposable
+    {
+        private IntPtr _tokenizerHandle;
+        private bool _disposed = false;
+
+        public Tokenizer(Model model)
+        {
+            Result.VerifySuccess(NativeMethods.OgaCreateTokenizer(model.Handle, out _tokenizerHandle));
+        }
+
+        public Sequences EncodeBatch(string[] strings)
+        {
+            IntPtr nativeSequences = IntPtr.Zero;
+            IntPtr stringArray = IntPtr.Zero;
+            Result.VerifySuccess(NativeMethods.OgaCreateStringArray(out stringArray));
+            try
+            {
+                foreach (string s in strings)
+                {
+                    Result.VerifySuccess(NativeMethods.OgaStringArrayAddString(stringArray, Utils.ToUtf8(s)));
```

Comment:
This produces a temporary `byte[]` which is garbage. 
Here is a suggestion that works best:
1) Calculate the length of the buffer required for the UTF-8 string, see example in the OrtValue API
2) Make an API that allocates a buffer for that string (May be the same API, but change the signature): essentially calls std::string::resize() and return a ptr to that buffer
3) Copy convert the string straight into the native buffer.